bot/cogs/hacktoberstats.py

```diff
@@ -0,0 +1,184 @@
+        brief="Get a user's Hacktoberfest contribution stats",
+    )
+    async def getstats(self, ctx, *args):
+        """
+        Query GitHub's API for PRs created by a GitHub user during the month of October that 
+        do not have an 'invalid' tag
+
+        For example:
+            !getstats heavysaturn
+
+        If a valid username is provided, an embed is generated and posted to the channel
+
+        Otherwise, post a helpful error message
+
+        The first input argument is treated as the username, any additional inputs are discarded
+        """
+        username = args[0]
+        PRs = await self.getoctoberprs(username)
+
+        postchannel = self.bot.get_channel(self.channelID)
+        if PRs:
+            statsembed = self.buildembed(username, PRs)
+            await postchannel.send('Here are some stats!', embed=statsembed)
+        else:
+            await postchannel.send(f"No October GitHub contributions found for '{username}'")
+
+    def buildembed(self, username: str, PRs: typing.List[dict]) -> discord.Embed:
+        """
+        Return a stats embed built from username's PRs
+        """
+        PRstats = self._summarizePRs(PRs)
+        
+        n = PRstats['nPRs']
+        if n >= 5:
+            shirtstr = f"**{username} has earned a tshirt!**"
+        elif n == 4:
+            shirtstr = f"**{username} is 1 PR away from a tshirt!**"
+        else:
+            shirtstr = f"**{username} is {5 - n} PRs away from a tshirt!**"
+
+        statsembed = discord.Embed(
+            title=f"{username}'s Hacktoberfest'",
+            color=discord.Color(0x9c4af7),
+            description=f"{username} has made {n} {Stats.contributionator(n)} in October\n\n{shirtstr}\n\n"
+            )
+        statsembed.set_thumbnail(url=f"https://www.github.com/{username}.png")
+        statsembed.set_author(
+            name="Hacktoberfest",
+            url="https://hacktoberfest.digitalocean.com", 
+            icon_url="https://hacktoberfest.digitalocean.com/assets/logo-hacktoberfest.png"
+            )
+        statsembed.add_field(
+            name="Top 5 Repositories:", 
+            value=self._buildtop5str(stats)
+            )
+        
+        return statsembed
+
+    @staticmethod
+    async def getoctoberprs(username: str) -> typing.List[dict]:
+        """
+        Query GitHub's API for PRs created during the month of October by username that do 
+        not have an 'invalid' tag
+        
+        If PRs are found, return a list of dicts with basic PR information
+        
+        For each PR:
+        
+            {
+            repo_url: str
+            repo_shortname: str (e.g. "discord-python/hacktoberbot")
+            created_at: datetime.datetime
+            }
+        
+        Otherwise, return None
+        """
+        queryURL = f"https://api.github.com/search/issues?q=-label:invalid+type:pr+is:public+author:{username}+created:2018-10-01..2018-10-31&per_page=300"
```

Comment:
Can this be split into multiple lines? maybe one new line for each new filter on the search.

Reply:
Something like:

```python
base_url = "https://api.github.com/search/issues?q="
        not_label = "invalid"
        action_type = "pr"
        is_query = f"public+author:{username}"
        date_range = "2018-10-01..2018-10-31"
        per_page = "300"
        query_url = f"{base_url}-label:{not_label}+type:{action_type}+is:{is_query}+created:{date_range}&per_page={per_page}"
```

Or should I go with a concatenated multi-line string?